apps/explorer/lib/explorer/migrator/shrink_internal_transactions.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Explorer.Migrator.ShrinkInternalTransactions do
+  @moduledoc """
+  Removes the content of input and output fields in internal transactions.
+  This migration is disabled unless SHRINK_INTERNAL_TRANSACTIONS_ENABLED env variable is set to true.
+  """
+
+  use Explorer.Migrator.FillingMigration
+
+  import Ecto.Query
+
+  alias Explorer.Chain.{Block, InternalTransaction}
+  alias Explorer.Migrator.FillingMigration
+  alias Explorer.Repo
+
+  @migration_name "shrink_internal_transactions"
+
+  @impl FillingMigration
+  def migration_name, do: @migration_name
+
+  @impl FillingMigration
+  def last_unprocessed_identifiers(%{max_block_number: -1} = state), do: {[], state}
+
+  def last_unprocessed_identifiers(%{max_block_number: from_block_number} = state) do
+    limit = batch_size() * concurrency()
+    to_block_number = max(from_block_number - limit + 1, 0)
+
+    {Enum.to_list(from_block_number..to_block_number), %{state | max_block_number: to_block_number - 1}}
+  end
+
+  def last_unprocessed_identifiers(state) do
+    query =
+      from(
+        it in InternalTransaction,
+        where: fragment("length(?) > 4", it.input) or not is_nil(it.output),
+        select: max(it.block_number)
+      )
+
+    max_block_number = Repo.one(query, timeout: :infinity)
+
+    state
+    |> Map.put(:max_block_number, max_block_number || -1)
+    |> last_unprocessed_identifiers()
+  end
+
+  @impl FillingMigration
+  def unprocessed_data_query, do: nil
+
+  @impl FillingMigration
+  def update_batch(block_numbers) do
+    block_hashes_query = from(b in Block, where: b.number in ^block_numbers, select: b.hash)
+    block_hashes = Repo.all(block_hashes_query)
+
+    query =
+      from(it in InternalTransaction,
+        where: it.block_hash in ^block_hashes,
```

Comment:
Can we match by `block_number` here, to avoid an extra query to the `blocks` table?  
```suggestion
        where: it.block_number in ^block_numbers,
```

Reply:
`block_hash` was used there on purpose because there is no index on `block_number` field

Reply:
Got it. Then does it makes sense to write it as a subquery? I think it will work faster if we won't load all block hashes into app's memory.

Reply:
Yep, processed in 5b1ad6d432e53189f7e1f578ec0d25bda689960b